sysdump/sysdump.go

```diff
@@ -639,10 +639,23 @@ func (c *Collector) Run() error {
 		},
 	}
 
+	// Adjust the worker count to make enough headroom for tasks that submit sub-tasks.
+	// This is necessary because 'Submit' is blocking.
+	wc := 1
+	for _, t := range tasks {
+		if t.CreatesSubtasks && !c.shouldSkipTask(t) {
+			wc++
+		}
+	}
+	// Take the maximum between the specified worker count and the minimum number of workers required.
+	wc = int(math.Max(float64(wc), float64(c.options.WorkerCount)))
```

Comment:
I think we can avoid the `int -> float64 -> int` conversion and `math` import here:

```suggestion
	if wc < c.options.WorkerCount {
		wc = c.options.WorkerCount
	}
```

Reply:
Indeed — thanks @tklauser 🚀